tests/unit/via/services/youtube_transcript_test.py

```diff
@@ -0,0 +1,300 @@
+                }
+            ),
+            Any.instance_of(TranscriptInfo).with_attrs(
+                {
+                    "language_code": caption_tracks[1]["languageCode"],
+                    "autogenerated": False,
+                    "name": caption_tracks[1]["name"]["simpleText"],
+                    "url": caption_tracks[1]["baseUrl"],
+                }
+            ),
+        ]
+
+    def test_get_transcript_infos_error_response(self, svc, http_service):
+        # We get an error response from the YouTube API.
+        http_service.post.side_effect = UnhandledUpstreamException(
+            "Something went wrong"
+        )
+
+        with pytest.raises(UnhandledUpstreamException):
+            svc.get_transcript_infos("test_video_id")
+
+    @pytest.mark.parametrize(
+        "response_body,exception_class",
+        [
+            (b"foo", JSONDecodeError),  # Not valid JSON.
+            (b"[]", TypeError),  # Not a dict.
+            (b"{}", KeyError),  # No "captions" key.
+            (b'{"captions": 23}', TypeError),  # "captions" isn't a dict.
+            (b'{"captions": {}}', KeyError),  # No "playerCaptionsTracklistRenderer".
+            # "playerCaptionsTracklistRenderer" isn't a dict.
+            (
+                b'{"captions": {"playerCaptionsTracklistRenderer": 23}}',
+                TypeError,
+            ),
+            # No "captionTracks".
+            (
+                b'{"captions": {"playerCaptionsTracklistRenderer": {}}}',
+                KeyError,
+            ),
+            # "captionTracks" isn't a list.
+            (
+                b'{"captions": {"playerCaptionsTracklistRenderer": {"captionTracks": 23}}}',
+                TypeError,
+            ),
+            # No "languageCode".
+            (
+                b'{"captions": {"playerCaptionsTracklistRenderer": {"captionTracks": [{"name": {"simpleText": "English"}, "baseUrl": "https://example.com"}]}}}',
+                KeyError,
+            ),
+            # No "name".
+            (
+                b'{"captions": {"playerCaptionsTracklistRenderer": {"captionTracks": [{"languageCode": "en", "baseUrl": "https://example.com"}]}}}',
+                KeyError,
+            ),
+            # "name" isn't a dict.
+            (
+                b'{"captions": {"playerCaptionsTracklistRenderer": {"captionTracks": [{"languageCode": "en", "name": 23, "baseUrl": "https://example.com"}]}}}',
+                TypeError,
+            ),
+            # No "simpleText".
+            (
+                b'{"captions": {"playerCaptionsTracklistRenderer": {"captionTracks": [{"languageCode": "en", "name": {}, "baseUrl": "https://example.com"}]}}}',
+                KeyError,
+            ),
+            # No "baseUrl".
+            (
+                b'{"captions": {"playerCaptionsTracklistRenderer": {"captionTracks": [{"languageCode": "en", "name": {"simpleText": "English"}}]}}}',
+                KeyError,
+            ),
```

Comment:
I put in tests for quite a few (all?) of the ways that `get_transcript_infos()` can crash on unexpected data. Maybe a bit excessive? But it might come in handy if we ever decide we want to bullet-proof this function.

This still doesn't cover everything that can go wrong, because it only covers crashes. If the `language_code`, `name` or `baseUrl` is present but isn't a string then it'll return `TranscriptInfo`'s whose `language_code`, `name` or `url`'s aren't strings. This would probably fail later on when code elsewhere tries to save these non-strings in the DB.

Or if `captionTracks` is an empty list then it'll return an empty list.